### PR TITLE
CCD-4198 Find the list of affected cases

### PR DESCRIPTION
**Pull request by rafalkalita**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CCD-4198

### Change description ###
Following supplementary data prod issue have created an endpoint to search for cases with HMCTSServiceID, but no orgs_assigned_users present in supplementary_data.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

**Review by mattnayler — COMMENTED** [review]
FYI: Couple of minor flags in Maintainability section of sonar report.

File: src/main/java/uk/gov/hmcts/ccd/ApplicationParams.java [hunk 1]
```diff
@@ -610,4 +613,8 @@ public Integer getRequestScopeCachedCaseTypesTillHour() {
     public Integer getSystemUserTokenCacheTTLSecs() {
         return systemUserTokenCacheTTLSecs;
     }
+
+    public List<String> getInvalidSupplementaryDataCaseTypes() {
+        return invalidSupplementaryDataCaseTypes;
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/ccd/ApplicationParams.java:618]: Resolved in 36fcfe029df2f178341d63c787066f74dfad7c82

File: src/main/java/uk/gov/hmcts/ccd/v2/external/controller/InvalidCaseSupplementaryDataController.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                                      List<CaseAssignedUserRole> caseAssignedUserRoles) {
+
+        Long invalidCaseId = invalidSupplementaryDataCase.getCaseId();
+        Optional<CaseAssignedUserRole> userRole = caseAssignedUserRoles.stream()
+            .filter(e -> invalidCaseId.toString().equals(e.getCaseDataId()))
+            .findFirst();
+
+        if (userRole.isPresent()) {
+            invalidSupplementaryDataCase.setUserId(userRole.get().getUserId());
+            invalidSupplementaryDataCase.setCaseRole(userRole.get().getCaseRole());
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/ccd/v2/external/controller/InvalidCaseSupplementaryDataController.java]: Resolved as agreed by adding comments in code in 36fcfe029df2f178341d63c787066f74dfad7c82

File: src/test/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepositoryTest.java [hunk 3]
```diff
@@ -681,6 +683,98 @@ public void findByParamsWithLimit_shouldFilterOnCaseType() {
...
+        assertEquals(3, response.size());
+        assertTrue(response.stream().anyMatch(cd -> cd.getReference().equals(1504259907311111L)));
+        assertTrue(response.stream().anyMatch(cd -> cd.getReference().equals(1504259907311112L)));
+        assertTrue(response.stream().anyMatch(cd -> cd.getReference().equals(1504259907311113L)));
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts =
+        {"classpath:sql/insert_cases_supplementary_data.sql"})
+    public void findCasesFilteredByDateFrom() {
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepositoryTest.java]: Resolved in 36fcfe029df2f178341d63c787066f74dfad7c82

File: src/main/java/uk/gov/hmcts/ccd/v2/external/controller/InvalidCaseSupplementaryDataController.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+        List<InvalidCaseSupplementaryDataItem> invalidSupplementaryDataCases = invalidSupplementaryDataOperation
+            .getInvalidSupplementaryDataCases(request.getDateFrom(), request.getDateTo(), request.getLimit());
+
+        if (request.getSearchRas() != null && request.getSearchRas()) {
+            List<Long> casesList = invalidSupplementaryDataCases.stream()
+                .map(InvalidCaseSupplementaryDataItem::getCaseId).collect(Collectors.toList());
+
+            List<CaseAssignedUserRole> caseAssignedUserRoles = this.caseAssignedUserRolesOperation
+                .findCaseUserRoles(casesList, emptyList());
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/ccd/v2/external/controller/InvalidCaseSupplementaryDataController.java:92]: Resolved by adding case roles filter in 36fcfe029df2f178341d63c787066f74dfad7c82

**Review by mattnayler — APPROVED** [review]
LGTM

**Review by skandala-cgi — APPROVED** [review]
GTM

**Pull request closed without merging**